html/modules/custom/reliefweb_import/src/Service/WorkDayJobImporter.php

```diff
@@ -0,0 +1,326 @@
+
+  /**
+   * The settings for the WorkDay service.
+   *
+   * @var array
+   */
+  protected array $settings;
+
+  /**
+   * Set the settings for the WorkDay service.
+   *
+   * @param array $settings
+   *   The settings to use for the WorkDay service.
+   */
+  public function setSettings(array $settings): void {
+    $this->settings = $settings;
+  }
+
+  /**
+   * Get authorization token from WorkDay.
+   */
+  public function getAuthToken(): string {
+    $timeout = $this->settings['timeout'] ?? 10;
+    $base_url = $this->settings['base_url'] ?? '';
+    $tenant = $this->settings['tenant'] ?? '';
+    $client_id = $this->settings['client_id'] ?? '';
+    $client_secret = $this->settings['client_secret'] ?? '';
+    $refresh_token = $this->settings['refresh_token'] ?? '';
+    $token_url = $base_url . '/ccx/oauth2/' . $tenant . '/token';
+
+    $response = $this->httpClient->request('POST', $token_url, [
+      'connect_timeout' => $timeout,
+      'timeout' => $timeout,
+      'headers' => [
+        'Content-Type' => 'application/x-www-form-urlencoded',
+        'Accept' => 'application/json',
+      ],
+      'form_params' => [
+        'grant_type' => 'refresh_token',
+        'client_id' => $client_id,
+        'client_secret' => $client_secret,
+        'refresh_token' => $refresh_token,
+      ],
+    ]);
+
+    if ($response->getStatusCode() !== 200) {
+      throw new \Exception('Failure with response code: ' . $response->getStatusCode());
+    }
+
+    try {
+      $token = json_decode($response->getBody()->getContents(), TRUE, flags: \JSON_THROW_ON_ERROR)['access_token'];
+    }
+    catch (\Exception $e) {
+      throw new \Exception('Unable to decode the auth token: ' . $e->getMessage());
+    }
+
+    return $token;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function importJobs(int $limit = 50): void {
+    $jobs = $this->getDocuments($limit);
+
+    // Switch to proper user and import XML.
+    $uid = $this->settings['uid'] ?? 2;
+
+    /** @var \Drupal\user\UserInterface $account */
+    $account = $this->entityTypeManager->getStorage('user')->load($uid);
+    $account->addRole('job_importer');
+    $this->accountSwitcher->switchTo($account);
+
+    $this->importDocuments($jobs);
+
+    // Restore user account.
+    $this->accountSwitcher->switchBack();
+  }
+
+  /**
+   * Retrieve documents from the WorkDay.
+   *
+   * @param int $limit
+   *   Number of documents to fetch.
+   *
+   * @return array
+   *   List of documents keyed by IDs.
+   */
+  public function getDocuments(int $limit = 50): array {
+    $this->getLogger()->info('Retrieving documents from the WorkDay.');
+
+    $documents = [];
+
+    try {
+      $timeout = $this->settings['timeout'] ?? 10;
+      $base_url = $this->settings['base_url'] ?? '';
+      $tenant = $this->settings['tenant'] ?? '';
+      $url = $base_url . '/ccx/api/recruiting/v4/' . $tenant . '/jobPostings';
```

Comment:
Same comment as before, if `base_url` or `tenant` are not present, the URL will likely be invalid. Maybe there should be a method to validate the config and bail out if not.